### ISSUE TITLE
Enhancements to the family members (personen) component

### ISSUE THREAD

**Issue by joeridiederen** [open-formulieren/open-forms#4925]
### Thema / Theme

Frontend

### Omschrijving / Description

This RFC proposes enhancements to the existing “personen” component within Open Forms. The component leverages a citizen service number (BSN) of the authenticated individual to query a backend registry (via StUF-BG or HaalCentraal) and retrieve person-related data. 

Initially, it will focus on four key scenarios: 

- retrieving children, 
- retrieving a partner, 
- retrieving housemates, and 
- retrieving a specific person. 

Each scenario can return a consistent set of _core variables_ — full name, BSN, birthdate.

The “personen” component will integrate closely with HaalCentraal’s standardized fields and filtering capabilities. For the first three scenarios (children, partner, housemates), the component retrieves a curated list of individuals based on configured filters and returns the core variables once the user selects a single result. Additionally, these scenarios may provide counts (e.g., the number of children or housemates). For the “Haal Persoon Op” scenario, administrators can return not only the core variables but also additional attributes selected from HaalCentraal’s [fields tool](https://brp-api.github.io/Haal-Centraal-BRP-bevragen/v2/fields).

**Scenario's**

**Retrieve Children**
Description: This scenario fetches individuals who are children of the authenticated person. Administrators can configure filters such as:
- Age-based filters (e.g., only show children older or younger than 18).
- Address-based filters (e.g., only children living at the same address).
- Deceased status (e.g., exclude or include deceased children).
- The component can also return the total count of filtered children (e.g., “You have 3 children that match this criteria”). Once a user selects one child, the component provides the core variables.
Purpose: Validate relationships and ages, prefill child-related fields, or trigger age-based logic in the form (e.g., determining eligibility for certain services). The count of children can inform the user or trigger logic based on family size.

**Retrieve partner**
This scenario lists the partner(s) of the authenticated individual. Filters include:
- Address-based filters (e.g., only show a partner if they live at the same address).
- Upon selection of the partner, the component returns the core variables.
Purpose: Facilitate auto-filling partner details for joint applications, verify relationship-based eligibility criteria, or streamline subsequent queries in the form.

**Retrieve housemates**
This scenario focuses on retrieving individuals who share the same address (housemates). The component filters based on address data and can return a count of how many housemates are present (e.g., “6 housemates found”). Once a housemate is selected, the component provides the core variables.
Purpose: Validate shared occupancy details, support applications that require information about co-residents, or enable conditional logic based on household composition. The count can inform whether the user must provide additional information or documentation.

**Retrieve person**
Description: This scenario allows an administrator to retrieve a specific individual by BSN. In addition to the default core variables, administrators can select from the full range of attributes offered by HaalCentraal’s [fields tool](https://brp-api.github.io/Haal-Centraal-BRP-bevragen/v2/fields). No additional count is required here. This flexible configuration enables advanced customization of the returned data.
Purpose: Handle complex form logic where detailed person attributes are required without additional steps or manual API calls. This scenario supports richer data retrieval for more complex workflows.

### Added value / Toegevoegde waarde

- Default mappings and scenario-based filtering lower the barrier to entry. The “Haal Persoon Op” scenario provides the option to go deeper by customizing the retrieval of extended attributes when needed.
- Administrators can enrich the retrieved data using HaalCentraal’s fields tool, selecting additional attributes that serve the form’s specific requirements.
- Age, address, and deceased-status filters, along with the ability to present counts, ensure the displayed individuals are directly relevant and informative to the user’s context.

### Aanvullende opmerkingen / Additional context

For the initial MVP, each instance of the “personen” component **can** be limited to a single selection. This design choice simplifies both development and the user experience. By starting small, we ensure that when one individual is selected, Open Forms reliably creates the specified core variables (e.g., eigenschapsnaam.fullname, eigenschapsnaam.BSN, eigenschapsnaam.birthdate) without additional complexity. These variables can be referenced in the form’s logic even before the user makes a selection, enabling a smooth “happy flow.” For example, administrators can write logic anticipating personen.fullname or personen.BSN and apply conditions or display fields once these variables become populated.

But the end goal is to allow dynamic variable creation for multiple selections. Advice on the feasibility of this is desired.

As a future-proof perspective, there are two possible approaches. The first is a static model with a predefined maximum number of options, where you establish an upper limit (e.g., five children) and set up corresponding variables and repeating groups in advance. This makes development simpler and predictable, but can feel constrained if real-world data often exceeds these limits. The second approach is a truly dynamic model that treats arrays and loops as first-class citizens, allowing the form to expand or contract in real-time without predefined boundaries. This approach provides maximum flexibility and future-proof scalability, but demands a more complex implementation and a steeper learning curve for administrators.

**Comment by sergei-maertens** [comment]
Refinement: analysis of the impact is needed before we can make an estimate.

**Comment by sergei-maertens** [comment]
Notities refinement

# npFamilyMembers component

Epic: https://github.com/open-formulieren/open-forms/issues/4944

## Wie is wie

* Menno en Peter -> richten alle formulieren in bij Utrecht
* Raoul: informatieprocesmanager en PO voor OF Utrecht

## Onderwerp: ophalen van familieleden

* Zit nu ophalen gezindsleden in

* Levert selectbox op waarin personen worden weergegeven (naam + bsn)

* Bij Werk en Inkomen

    - moeten partner en kinderen weten
    - als je met DigiD inlogt -> kan je ophalen
    - kan je ook ongeregistreerd samenwonen -> handmatig partner etc. toevoegen

* Bestaande oplossing "gezinspagina"

    1. Inloggen met DigiD
    2. Basis met Gezin

        - Persoonsgegevens (BSN, dob etc.)
        - Contactgegevens

    3. Gezinspagina

        - Partnergegevens (BSN, voorletters, tussenvoegsel, geboortedatum(!))
        - Kinderen: tabel met kolommen **BSN - voornamen - geboortedatum**. Gelimiteerd
          tot kinderen op hetzelfde adres (? waar ligt de filter, in BRP klinkt onlogisch).
        - Kunnen filteren op condities (leeftijd, kind vs zelfstandig...)
        - Opgeven voor basisschool -> filter op kinderen die 3+ jaar zijn
        - Overleden kinderen filteren (geldt niet voor partner?)

       -> is prefill mechanisme

* Huisgenoten ophalen -> privacytechnisch kan dat niet (misschien x aantal huisgenoten)

* Bijv. bijstandsuitkering -> per kind aangeven of die naar school gaat, kinderbijslag
  etc.

* Welke gegevens hebben van kinderen?

    - BSN
    - geboortedatum
    - voornamen/tussenvoegsel/achternaam
    - leeftijd -> verrekenen aan onze kant (HC heeft het wel al, als't goed is)
    - geen extra velden (onder water)

**Comment by knechtj** [comment]
@joeridiederen 
What is the status of the Family Member component using StUF-BG issue? This is logged by me in the Dimpact servicedesk tool.
If the issue here is unknown, please let me know, then I will provide it here.
I am not sure if GitHUB and this thread is the right way to mention this bug.

**Comment by joeridiederen** [comment]
> [@joeridiederen](https://github.com/joeridiederen) What is the status of the Family Member component using StUF-BG issue? This is logged by me in the Dimpact servicedesk tool. If the issue here is unknown, please let me know, then I will provide it here. I am not sure if GitHUB and this thread is the right way to mention this bug.

I think that's a separate issue. Regarding placement: from Rotterdam, we've agreed with Dimpact to post issues both here and on the Dimpact JIRA, because refinement, realization, and monitoring ultimately take place here. Additionally, there's more opportunity for discussion here, as the community is larger and somewhat more advanced. So I'd definitely post it on both platforms, including the correct reference.

**Comment by sergei-maertens** [comment]
The new children and partners family members components facilitate a bunch of these features, including:

* filtering on deceased/alive status
* age-based filters

The returned data type is an array, which has an intrinsic `length` property so you can derive counts. We're also working on functionality that you can use this information (with indeed, a fixed set of properties like BSNs, names, date of birth...) as seed data for the repeating group component so that you can enter additional information for each child.

Some notable exclusions:

* house mates - this was brought up in a refinement with another party and they confirmed our reservations towards GDPR requirements. Being able to retrieve or expose this information sounds a lot like a AVG/GDPR/privacy violation. This is currently not on on our roadmap
* similar concerns apply to being able to lookup any person via an (arbitrary) BSN

I'm going to decouple this issue from the epic that we're working on an leave it in the backlog for a while longer so you have a chance to address this comment, but personally I'm leaning towards a "wontfix/not planned" for the mentioned exclusions. Iterating on the partners/children components is of course possible within reason.

**Comment by sergei-maertens** [comment]
Closing this per my above comment - I think the functional needs are covered through the new Family members components (children, partners). You can of course create feature requests/enhancements if you run into any more shortcomings in those components.
